library/core/src/ptr/mod.rs

```diff
@@ -210,7 +410,7 @@ pub unsafe fn drop_in_place<T: ?Sized>(to_drop: *mut T) {
 #[rustc_const_stable(feature = "const_ptr_null", since = "1.24.0")]
 #[rustc_diagnostic_item = "ptr_null"]
 pub const fn null<T>() -> *const T {
-    0 as *const T
+    invalid::<T>(0)
```

Comment:
```suggestion
    invalid(0)
```
Is there a reason the type is explicitly specified (same for below)?

Reply:
By default I prefer to avoid return-type-inference in raw pointer stuff since the cost-benefit is Not Great. Certainly it's fine here, just a habit.

Reply:
I prefer to avoid inference with raw pointers most of the time too, just wondered because the type is specified explicitly in the return type. Mentioning it is fine, though.